src/contract.ts

```diff
@@ -0,0 +1,38 @@
+import assert from "assert";
+import fs from "fs";
+import Web3 from "web3";
+import type { AbiItem } from "web3-utils";
+import { isAddress } from "web3-utils";
+import { Contract } from "web3-eth-contract";
+import path from "path";
+import { AccountIngress } from "../types/@types/AccountIngress"; 
+  
+declare const CONFIG: ConfigType;
+
+export class AllowedListContract {
+    web3: Web3;
+    ingressContract: AccountIngress;
+    allowedListContract!: Contract; // it can be undefined at first, cause if constructor fails it will never be initialized
+
+    constructor() {
+        this.web3 = new Web3(CONFIG.sidechain.nodeUrl);
+        const accountIngress = JSON.parse(fs.readFileSync(path.resolve(__dirname, CONFIG.sidechain.accountIngress), "utf-8"));
+        this.ingressContract = new this.web3.eth.Contract(accountIngress["abi"] as AbiItem[], CONFIG.sidechain.address) as unknown as AccountIngress;
+    }
+
+    public async initializeContract(): Promise<AllowedListContract> {
```

Comment:
Why is this separate method, not just code in constructor?

Reply:
Constructor shouldn't be async (using awaits requires that). I mean it is possible, but usually the approach is to introduce function for this and then call it.